streaming/stream_session.cc

```diff
@@ -217,7 +243,7 @@ void stream_session::init_messaging_service_handler() {
                             });
                         },
                         cf.stream_in_progress()
-                    ).then_wrapped([s, plan_id, from, sink, estimated_partitions] (future<uint64_t> f) mutable {
+                    ).then_wrapped([s, plan_id, from, sink, estimated_partitions, cmd_status] (future<uint64_t> f) mutable {
```

Comment:
Is this capture (and the one below) needed?

Reply:
cmd_status must live until mutation_writer::distribute_reader_and_consume_on_shards is finished. The capture extends the lifetime.

Reply:
`do_with()` would make it clear what `cmd_status` lifetime should be.

Reply:
@asias Why does it have to live until distribute_reader_and_consume_on_shards is finished? The continuations which use it already capture it.

Reply:
Because the get_next_mutation_fragment which uses the cmd_status is called again and again within distribute_reader_and_consume_on_shards. 

A second thought: since get_next_mutation_fragment is stored inside multishard_writer and we have do_with to keep the lifetime of multishard_writer. I think capture cmd_status in the get_next_mutation_fragment alone is enough.

```
191 future<uint64_t> distribute_reader_and_consume_on_shards(schema_ptr s,
192     dht::i_partitioner& partitioner,
193     flat_mutation_reader producer,
194     std::function<future<> (flat_mutation_reader)> consumer,
195     utils::phased_barrier::operation&& op) {
196     return do_with(multishard_writer(std::move(s), partitioner, std::move(producer), std::move(consumer)), std::move(op), [] (multishard_writer& writer, utils::phased_barrier::operation&) {
197         return writer();
198     });
199 }
```

Reply:
As long as `get_next_mutation_fragment` is called again and again, it must be alive, and as long as its alive, its captures are alive as well.